task/download-sbom-from-url-in-attestation/0.1/download-sbom-from-url-in-attestation.yaml

```diff
@@ -0,0 +1,256 @@
+        echo 'Initializing TUF root...'
+        cosign initialize --mirror "${TUF_MIRROR}" --root "${TUF_MIRROR}/root.json"
+      fi
+
+      for image in "$@"; do
+        echo "Getting attestation for $image"
+        mkdir -p "$WORKDIR/$image"
+        cosign verify-attestation \
+          --type slsaprovenance \
+          "${cosign_args[@]}" \
+          "$image" > "$WORKDIR/$image/attestation.json"
+      done
+    args:
+    - $(params.IMAGES[*])
+
+  - name: do-magic
+    image: quay.io/redhat-appstudio/appstudio-utils:5bd7d6cb0b17f9f2eab043a8ad16ba3d90551bc2@sha256:8c7fcf86af40c71aeb58e4279625c8308af5144e2f6b8e28b0ec7e795260e5f7
+    script: |
+      #!/usr/bin/env bash
+      set -o errexit -o nounset -o pipefail
+
+      get_from_www_auth_header() {
+          local www_authenticate=$1
+          local key=$2
+          # shellcheck disable=SC2001
+          # E.g.
+          #   www_authenticate='Bearer realm="https://ghcr.io/token",service="ghcr.io"'
+          #   key=service
+          #   -> ghcr.io
+          sed "s/.*$key=\"\([^\"]*\)\".*/\1/" <<< "$www_authenticate"
+      }
+
+      get_container_auth() {
+          # https://man.archlinux.org/man/containers-auth.json.5
+
+          local image=$1
+
+          local runtime_dir="${XDG_RUNTIME_DIR:-}"
+          local config_home="${XDG_CONFIG_HOME:-$HOME/.config}"
+          if [[ -n "$runtime_dir" ]]; then
+              default_authfile="${runtime_dir}/containers/auth.json"
+          else
+              default_authfile="$config_home/containers/auth.json"
+          fi
+
+          local primary_auth_file="${REGISTRY_AUTH_FILE:-$default_authfile}"
+
+          local maybe_auth_files=(
+              "$primary_auth_file"
+              "$config_home/containers/auth.json"
+              "$HOME/.docker/config.json"
+              "$HOME/.dockercfg"
+          )
+          local auth_files=()
+          for file in "${maybe_auth_files[@]}"; do
+              if [[ -r "$file" ]]; then
+                  auth_files+=("$file")
+              fi
+          done
+
+          # registry.com/namespace/repo@sha256:digest -> registry.com/namespace/repo
+          local auth_key=${image%@*}
+
+          while true; do
+              for auth_file in "${auth_files[@]}"; do
+                  if jq -r -e --arg key "$auth_key" '.auths[$key].auth // empty' "$auth_file"; then
+                      echo "Found auth for $auth_key in $auth_file" >&2
+                      return 0
+                  fi
+              done
+
+              # Try less specific key, e.g. registry.com/namespace/repo -> registry.com/namespace
+              local new_key=${auth_key%/*}
+              if [[ "$new_key" = "$auth_key" ]]; then
+                  # Already tried all possible keys, no auth found
+                  echo "No auth found for $auth_key" >&2
+                  return 1
+              fi
+
+              auth_key=$new_key
+          done
+
+      }
+
+      download_blob() {
```

Comment:
This should absolutely not be done in Bash. We should use `oras blob fetch`. But we would have to productize oras.